standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HmsThriftHttpServlet.java

```diff
@@ -39,75 +48,119 @@ public class HmsThriftHttpServlet extends TServlet {
       .getLogger(HmsThriftHttpServlet.class);
 
   private static final String X_USER = MetaStoreUtils.USER_NAME_HTTP_HEADER;
-
   private final boolean isSecurityEnabled;
+  private final boolean jwtAuthEnabled;
+  public static final String AUTHORIZATION = "Authorization";
+  private JWTValidator jwtValidator;
+  private Configuration conf;
 
   public HmsThriftHttpServlet(TProcessor processor,
-      TProtocolFactory inProtocolFactory, TProtocolFactory outProtocolFactory) {
-    super(processor, inProtocolFactory, outProtocolFactory);
-    // This should ideally be reveiving an instance of the Configuration which is used for the check
+      TProtocolFactory protocolFactory, Configuration conf) {
```

Comment:
Adding a  new argument would break the compatibility/interoperability between HMS and clients. Instead, wouldn't it be ideal to add a new constructor with an added argument?

Reply:
@saihemanth-cloudera: HmsThriftHttpServlet is only used by HiveMetastoreServer if HTTP mode is enabled. Therefore HMSClients don't have anything to do with this class. 

However I totally agree with the point that we could have added a new constructor. I deliberately choose not to do so because this class was added very recently in Thrift over HTTP patch and  since the apis are still evolving, changing the signature of existing constructor made more sense to me. 

Let me know your thoughts.

Reply:
@sourabh912 - I didn't know that this class was recently introduced in the thrift over http patch. In that case, it would be ideal to leave this as such with the existing constructor. We can always do constructor overloading in the future if any use case arises.